adapter/internal/discovery/xds/server.go

```diff
@@ -266,14 +267,21 @@ func UpdateAPI(vHost string, apiProject mgw.ProjectAPI, environments []string) (
 		}
 	}()
 
-	if len(environments) == 0 {
-		environments = []string{config.DefaultGatewayName}
+	// If number of environments is zero, then the API would be deployed to the default Environment under systemHost
+	if len(deployedEnvironments) == 0 {
+		deployedEnvironments = []*synchronizer.GatewayLabel{
+			{
+				Name:           "Default",
```

Comment:
Shouldn't this be `config.DefaultGatewayName`?
```suggestion
				Name:           config.DefaultGatewayName,
```